### PR TITLE
Completed mock data generator & seeding script

### PR DESCRIPTION
**Pull request by edenlhn**
@pickles27 @jordanstholmes @KStockley  If one of you guys could review this that would be great! Thanks!

### CODE REVIEW

File: database/index.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+const Sequelize = require('sequelize');
+const db = new Sequelize('rewards', 'root', '', {
+  dialect: 'mysql'
+});
+
+const Reward = db.define('Reward', {
```
Comment (jordanstholmes) [inline database/index.js:6]: Primary key?

File: database/seed.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+const Promise = require('bluebird');
+
+const seedRewards = () => {
+  const rewards = [];
+  let tiers;
+  for (let i = 1; i < 101; i++) {
+
+    if (i % 10 === 0 && i % 8 !== 0) {
+      tiers = [1, 5, 10, 25, 50, 75, 100, 250, 500, 750];
+    } else if (i % 8 === 0) {
```
Comment (jordanstholmes) [inline database/seed.js]: If you put this `else if (i % 8 === 0) {` first, you don't need the `i % 8 !== 0` in that first "if"

File: database/seed.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        isLimited = true;
+        limitCount = 30;
+        backers = faker.random.number(30);
+      }
+      if (j === 9) {
+        isLimited = true;
+        limitCount = 10;
+        backers = faker.random.number(10);
+      }
+
```
Comment (jordanstholmes) [inline database/seed.js:37]: In the above section, I can't really tell what you're doing because I don't know what the significance of all the number literals are. Why 9? Why 10? Why 30? I would suggest either some named constants or else comments next to the lines where a number literal is used, to explain what it's representing. I think I'd tend towards the named constant, because then you could easily adjust your specifications all throughout the file by just changing that one variable.

File: database/index.js (hunk 1 above)
Comment (edenlhn) [reply database/index.js:6]: @jordanstholmes Sequelize automatically defaults to the id column if the primary key is not specified. Id column also not necessary to explicitly define in sequelize models.

File: database/seed.js (hunk 2 above)
Comment (edenlhn) [reply database/seed.js]: @jordanstholmes Will definitely make that change. Thanks!

File: database/seed.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  let rewardPromises = rewards.map((reward) => {
+    return db.Reward.create(reward)
+      .catch((err) => {
+        return Promise.resolve();
+      });
+  });
+
+  return Promise.all(rewardPromises);
+
```
Comment (jordanstholmes) [inline database/seed.js:63]: Is there a batchCreate or bulkCreate method that might let you take out the promises?

**Review by jordanstholmes — COMMENTED** [review]
Cool man :) Looks good. Clean clean clean. I only really have one major comment and that's the number literals. It was hard to follow your seed logic because I didn't know what the various numbers meant.

File: database/seed.js (hunk 3 above)
Comment (edenlhn) [reply database/seed.js:37]: @jordanstholmes There is absolutely no significance for the number literals. They're just a random number I chose so that every 10th or every 8th project will have a different set of reward tiers. The 10 and the 30 limit are also arbitrarily chosen numbers. The actual site would allow the projectCreator to manually set the number and level of reward tiers (and limit on certain reward tiers if applicable) but since I'm generating a bunch of mock data I just set an arbitrary trap to semi-randomize the data generation (without having completely random reward tier numbers that range all over the place). Maybe I'll add a comment above that section to clarify this point?

File: database/seed.js (hunk 4 above)
Comment (edenlhn) [reply database/seed.js:63]: @jordanstholmes I think there might be a bulkCreate method but I'm not sure. I'll have to double-check the Sequelize docs. I'll probably leave part as is for the time being and refactor to use bulkCreate at a later point if I have some extra time. Thanks so much for the feedback!

File: database/index.js (hunk 1 above)
Comment (jordanstholmes) [reply database/index.js:6]: Cool! Good to know :) Thanks

File: database/seed.js (hunk 3 above)
Comment (jordanstholmes) [reply database/seed.js:37]: Coolio. Me over here trying to figure out the pattern all "A Beautiful Mind" style, haha :D

File: database/seed.js (hunk 4 above)
Comment (jordanstholmes) [reply database/seed.js:63]: Fo sho, bro
